### ISSUE TITLE
Certain Product Blocks cause ordering issues for any post type blocks following it

### ISSUE THREAD

**Issue by ryanr14** [woocommerce/woocommerce-gutenberg-products-block#1195]
**Describe the bug**
This is a bit hard to condense to a title that makes sense, but if you use any of the three following blocks and then also use a block that gets posts and attempts to order them there are ordering issues.

Blocks causing the issue
Best Selling, Top Rated, and Newest Products.

**To Reproduce**
1. Add one of the above blocks to a page.
2. Right beneath that add a block that also queries posts, in my example I'm using an Automattic plugin called [Post List Block](https://wordpress.org/plugins/posts-list-block/).
3. Preview the two blocks and you'll notice the Post List shows your posts out of order.
4. If you move the Post List Block above the product block things work as they should.

**Screenshots**
https://cld.wthms.co/VxMFK

**Expected behavior**
No matter what is above or below the block it shouldn't effect how other blocks order things like posts, etc... :)

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by ryanr14** [comment]
Howdy @juliaamosova, I just tested again with both WooCommerce Blocks activated and deactivated and the issue exists still. If a post list block, like the Post Block List from Automattic, is put below one of three blocks mentioned above from WooCommerce core the post dates are out of order.

![https://cld.wthms.co/Xnfk1J](https://cld.wthms.co/Xnfk1J+)
Image link: https://cld.wthms.co/Xnfk1J

If you move the Post List block above those mentioned blocks the dates appear in correct chronological order from newest to oldest.

**Comment by mikejolley** [comment]
@ryanr14 Old version of blocks/Woo core? I have no issue with Blocks 2.5 - I looked back and can see why this may have been an issue in Blocks 2.3 however  - https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/v2.3.0/src/BlockTypes/AbstractProductGrid.php#L274-L275 The query args were not removed. They are now :)

**Comment by mikejolley** [comment]
Blocks 2.4 is also affected - this condition is not ran if the transient is used: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/v2.4.0/src/BlockTypes/AbstractProductGrid.php#L273

**Comment by mikejolley** [comment]
@nerrad Do we bother doing a 2.4.x release for core to include, or just leave this to be resolved naturally when Blocks 2.5 is released in core?

**Comment by nerrad** [comment]
tldr;  If this is an easy cherrypick, let's prep the release/2.4 for another release and cherrypick it in.  WC core has the flexibility to include it in the next release.

Longer answer:

Depends on impact:  Does it affect many or few?  Is it critical or minor?  Will it grow in impact or probably stay small in impact?  Also, we should consider how likely the fix is to introduce other problems.  Do we know what specifically in 2.5 fixes the issue?  Can it easily be cherrypicked into a 2.4.x release?

I don't know the answer to the impact questions in terms of how many this affects.  I see the Post List block is currently only active on 10 sites (does that include WP.com usage? I don't know).  However, if this is a fairly easy cherry-pick with minimal impact then would be nice to improve the experience for users of that block.  I wonder if this potentially has impact on other things as well.

**Comment by mikejolley** [comment]
It would need a specific 2.4 patch, not a cherry pick, since you only need to move the condition but we don't want to include the other refactors.

**Comment by nerrad** [comment]
Now that 2.5 has been released (and is pending to be included in WC3.9 I think I can go ahead and close this.
